shared/datastore_cloud.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2018 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+import (
+	"context"
+
+	"cloud.google.com/go/datastore"
+)
+
+type cloudKey struct {
+	key *datastore.Key
+}
+
+func (k cloudKey) IntID() int64 {
+	return k.key.ID
+}
+
+// NewCloudDatastore creates a Datastore implementation that is backed by a
+// standard cloud datastore client (i.e. not running in AppEngine standard).
+func NewCloudDatastore(ctx context.Context, client *datastore.Client) Datastore {
+	return cloudDatastore{
+		ctx:    ctx,
+		client: client,
+	}
+}
+
+type cloudDatastore struct {
+	ctx    context.Context
+	client *datastore.Client
+}
+
+func (d cloudDatastore) Context() context.Context {
+	return d.ctx
+}
+
+func (d cloudDatastore) NewQuery(typeName string) Query {
+	return cloudQuery{
+		query: datastore.NewQuery(typeName),
+	}
+}
+
+func (d cloudDatastore) GetAll(q Query, dst interface{}) ([]Key, error) {
+	keys, err := d.client.GetAll(d.ctx, q.(cloudQuery).query, dst)
+	cast := make([]Key, len(keys))
+	for i := range keys {
+		cast[i] = cloudKey{key: keys[i]}
+	}
+	return cast, err
+}
+
+type cloudQuery struct {
+	query *datastore.Query
+}
+
+func (q cloudQuery) Filter(filterStr string, value interface{}) Query {
+	return cloudQuery{query: q.query.Filter(filterStr, value)}
```

Comment:
Nit: specify (or don't specify) the only field name consistently in this file (and `datastore_ae.go`).

Reply:
Done.